infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerAdapter.java

```diff
@@ -1,51 +1,59 @@
 package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
 
-import lombok.Getter;
-
 import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
 
-public abstract class CorfuReplicationClusterManagerAdapter {
-    @Getter
-    CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService;
+/**
+ * This is the interface for CorfuReplicationSiteManager.
+ * Implementation of this should have following members:
+ * 1. corfuReplicationDiscoveryService that is needed to notify the cluster configuration change.
+ * 2. localEndpoint that has the local node information.
+ *
+ */
+public interface CorfuReplicationClusterManagerAdapter {
 
-    @Getter
-    TopologyConfigurationMsg topologyConfig;
+    /**
+     *   Set the corfuReplicationDiscoveryServiceMember
+     */
+     void setCorfuReplicationDiscoveryService(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService);
 
-    public void connect(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService) {
-        this.corfuReplicationDiscoveryService = corfuReplicationDiscoveryService;
-        start();
-    }
+     /**
+     * Set the localEndpoint
+     */
+    void setLocalEndpoint(String endpoint);
 
-    public synchronized TopologyConfigurationMsg fetchTopology() {
-        topologyConfig = queryTopologyConfig();
-        return topologyConfig;
-    }
+    // This is the currentTopology cached at the adapter.
+    public TopologyConfigurationMsg getTopologyConfig();
```

Comment:
redundant accesor in the interface, we can remove this one like in the others...

Reply:
Others will not be accessed outside the adapter, but this can be accessed outside. Also, it provides the api to get the currentTopology without contacting the upper layer.